src/target/riscv/riscv.c

```diff
@@ -1601,9 +1655,21 @@ static int riscv_trigger_detect_hit_bits(struct target *target, int64_t *unique_
 				break;
 			case CSR_TDATA1_TYPE_MCONTROL:
 				hit_mask = CSR_MCONTROL_HIT;
+				*need_single_step = true;
 				break;
 			case CSR_TDATA1_TYPE_MCONTROL6:
 				hit_mask = CSR_MCONTROL6_HIT0 | CSR_MCONTROL6_HIT1;
+				if (r->tinfo_version == CSR_TINFO_VERSION_0) {
+					*need_single_step = true;
+				} else if (r->tinfo_version == RISCV_TINFO_VERSION_UNKNOWN
+					|| r->tinfo_version == CSR_TINFO_VERSION_1) {
```

Comment:
@sunnyzhu-learning, I don't quite get it:
Why `RISCV_TINFO_VERSION_UNKNOWN` is handled the same way as `CSR_TINFO_VERSION_1` and not the `CSR_TINFO_VERSION_0`?

Shouldn't this be rewritten as:
```
				if (r->tinfo_version == CSR_TINFO_VERSION_0
					|| r->tinfo_version == RISCV_TINFO_VERSION_UNKNOWN) {
					*need_single_step = true;
				} else if (r->tinfo_version == CSR_TINFO_VERSION_1) {
```

Reply:
Considering `RISCV_TINFO_VERSION_UNKNOWN` equal to `tinfo` register is not implemented, and `mcontrol6` is implemented, so `RISCV_TINFO_VERSION_UNKNOWN` is handled the same way as `CSR_TINFO_VERSION_1`, this is my understand.